src/Command/DrupalUpdate/UpdateScript.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+class UpdateScript
+{
+  /**
+   * Ignore files and folders list.
+   * @var string[]
+   */
+  private $ignore_updates_files = [
+        '.gitignore','.htaccess','CHANGELOG.txt','sites',
+    ];
+
+  /**
+   * Drupal directory folder path.
+   * @var string
+   */
+  private $dir;
+
+  /**
+   * @param string $dir
+   */
+  public function setDir(string $dir): void {
+    if (!defined('DRUPAL_ROOT')) {
+      define('DRUPAL_ROOT', $dir);
+    }
+    $this->updateScriptUtility->setDrupalRootDirPath();
+    $this->dir = DRUPAL_ROOT . "/docroot";
+  }
+
+  /**
+   * @var CheckInfo
+   */
+  private $checkinfo_obj;
+
+  /**
+   * @var UpdateScriptUtility
+   */
+  private $updateScriptUtility;
+
+  /**
+   * UpdateScript constructor.
+   *
+   */
+  public function __construct() {
+    $this->checkinfo_obj = new CheckInfo();
+    $this->updateScriptUtility = new UpdateScriptUtility();
+    $this->setDir(getcwd());
+  }
+
+  /**
+   * @param array $latest_security_updates
+   */
+  public function updateAvailableUpdates(array $latest_security_updates) {
+    if(count($latest_security_updates)>1){
+      // if get any available then list out in tabular format.
+      $this->updateScriptUtility->printReleaseDetail($latest_security_updates);
+
+      // update current repo with updated code based on download and folder replace method on same path.
+      $this->updateScriptUtility->updateCode($latest_security_updates);
+
+      // remove the unwanted tar or downloaded files.
+      $this->updateScriptUtility->unlinktarfiles($latest_security_updates);
+    }else{
+      // If no updates found.
+      echo "\n Branch Upto date \n";
+    }
+  }
+
+  /**
+   * Scan the directory and sub directory recorsive.
+   * Get the list of all .info files list with path.
+   * @param string $dir
+   */
+  public function getInfoFilesList() {
```

Comment:
Seems like $dir should be passed as an argument here. Why should it be a property?